cmd/codegen/gencpp.go

```diff
@@ -1,476 +1,503 @@
-extern "C" {
-#endif
-
-`, prefix))
-
-	sbCpp.WriteString(fmt.Sprintf(`
-#include "%[1]s_wrapper.h"
-#include "%[1]s_structs_accessor.h"
-
-`, prefix))
-
-	for _, s := range structs {
-		for _, m := range s.Members {
-			if strings.Contains(m.Name, "[") || strings.Contains(m.Type, "(") || strings.Contains(m.Type, "union") {
-				continue
-			}
-
-			setterFuncName := fmt.Sprintf("%[1]s_Set%[2]s", s.Name, m.Name)
-			if funk.ContainsString(skipFuncNames, setterFuncName) {
-				continue
-			}
-
-			// Generate setter function
-			structAccessorFuncs = append(structAccessorFuncs, FuncDef{
-				Args: fmt.Sprintf("(%[1]s *%[2]s, %[3]s v)", s.Name, s.Name+"Ptr", m.Type),
-				ArgsT: []ArgDef{
-					{
-						Name: s.Name + "Ptr",
-						Type: s.Name,
-					},
-					{
-						Name: "v",
-						Type: m.Type,
-					},
-				},
-				FuncName:     setterFuncName,
-				Location:     "",
-				Constructor:  false,
-				Destructor:   false,
-				StructSetter: true,
-				Ret:          "void",
-			})
-
-			getterFuncName := fmt.Sprintf("%[1]s_Get%[2]s", s.Name, m.Name)
-			if funk.ContainsString(skipFuncNames, getterFuncName) {
-				continue
-			}
-
-			structAccessorFuncs = append(structAccessorFuncs, FuncDef{
-				Args: fmt.Sprintf("(%[1]s *%[2]s)", s.Name, "self"),
-				ArgsT: []ArgDef{
-					{
-						Name: "self",
-						Type: s.Name,
-					},
-				},
-				FuncName:     getterFuncName,
-				Location:     "",
-				Constructor:  false,
-				Destructor:   false,
-				StructSetter: false,
-				StructGetter: true,
-				Ret:          m.Type,
-			})
-
-			sbHeader.WriteString(fmt.Sprintf("extern void %[1]s_Set%[2]s(%[1]s *%[3]s, %[4]s v);\n", s.Name, m.Name, s.Name+"Ptr", m.Type))
-			sbHeader.WriteString(fmt.Sprintf("extern %[4]s %[1]s_Get%[2]s(%[1]s *%[3]s);\n", s.Name, m.Name, "self", m.Type))
-
-			sbCpp.WriteString(fmt.Sprintf("void %[1]s_Set%[2]s(%[1]s *%[3]s, %[4]s v) { %[3]s->%[2]s = v; }\n", s.Name, m.Name, s.Name+"Ptr", m.Type))
-			sbCpp.WriteString(fmt.Sprintf("%[4]s %[1]s_Get%[2]s(%[1]s *%[3]s) { return %[3]s->%[2]s; }\n", s.Name, m.Name, "self", m.Type))
-		}
-	}
-
-	sbHeader.WriteString(`
-#ifdef __cplusplus
-}
-#endif
-`)
-
-	headerPath := fmt.Sprintf("%s_structs_accessor.h", prefix)
-	headerFile, err := os.Create(headerPath)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create cpp file %v: %v", headerPath, err)
-	}
-
-	defer headerFile.Close()
-
-	_, err = headerFile.WriteString(sbHeader.String())
-	if err != nil {
-		return nil, fmt.Errorf("failed to write to file %v: %v", headerPath, err)
-	}
-
-	cppPath := fmt.Sprintf("%s_structs_accessor.cpp", prefix)
-	cppFile, err := os.Create(cppPath)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create cpp file %v: %v", cppPath, err)
-	}
-
-	defer cppFile.Close()
-
-	_, err = cppFile.WriteString(sbCpp.String())
-	if err != nil {
-		return nil, fmt.Errorf("failed to write to file %v: %v", cppPath, err)
-	}
-
-	return structAccessorFuncs, nil
-}
+package main
+
+import (
+	"fmt"
+	"os"

```

Comment:
It seems there is something wrong with formatting: `gofmt -w gencpp.go function_deffinitions.go`